src/schedule/edf_schedule.c

```diff
@@ -74,30 +70,18 @@ static void edf_scheduler_run(void *data)
 		    task->state != SOF_TASK_STATE_RUNNING)
 			continue;
 
-		edf_pdata = edf_sch_get_pdata(task);
+		deadline = task_get_deadline(task);
 
-		deadline = edf_pdata->deadline;
-
-		if (current >= deadline &&
-		    !(task->flags & SOF_SCHEDULE_FLAG_IDLE)) {
+		if (deadline == SOF_TASK_DEADLINE_NOW) {
```

Comment:
this change also makes .flags unused, maybe it's worth at least mentioning this in the commit message. Not hugely important though, just for easier understanding

Reply:
I've already noted this in the commit, which removes flags.